.github/workflows/linux.yml

```diff
@@ -398,9 +398,150 @@ jobs:
+          echo 'APT::Get::Fix-Broken "true";' | sudo tee -a /etc/apt/apt.conf.d/82-fix-broken > /dev/null
+          echo 'APT::Get::no-install-recommends "true";' | sudo tee -a /etc/apt/apt.conf.d/83-no-recommends > /dev/null
+
+      - name: Clone OpenVINO
+        uses: actions/checkout@v4
+        with:
+          repository: 'openvinotoolkit/openvino'
+          path: ${{ env.OPENVINO_REPO }}
+          submodules: 'true'
+          ref: ${{ env.OV_BRANCH}}
+
+      - name: Install build dependencies
+        run: |
+          sudo -E ${OPENVINO_REPO}/install_build_dependencies.sh
+          sudo apt-get install ccache
+
+      - name: Clone GenAI
+        uses: actions/checkout@v4
+        with:
+          path: ${{ env.GENAI_REPO }}
+          submodules: recursive
+
+      - name: Setup ccache
+        uses: actions/cache@v4
+        with:
+          # Should save cache only if run in the master branch of the base repo
+          # github.ref_name is 'ref/PR_#' in case of the PR, and 'branch_name' when executed on push
+          save-always: ${{ github.ref_name == 'master' && 'true' || 'false'  }}
+          path: ${{ env.CCACHE_DIR }}
+          key: ${{ runner.os }}-${{ runner.arch }}-ccache-ov-and-genai-${{ matrix.build-type }}-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-${{ runner.arch }}-ccache-ov-and-genai-${{ matrix.build-type }}
+
+      - name: Build openvino + genai
+        run: |
+          cmake -DOPENVINO_EXTRA_MODULES=../openvino.genai -DCPACK_ARCHIVE_COMPONENT_INSTALL=OFF \
+              -DENABLE_OV_ONNX_FRONTEND=OFF -DENABLE_OV_PADDLE_FRONTEND=OFF -DENABLE_OV_TF_FRONTEND=OFF -DENABLE_OV_TF_LITE_FRONTEND=OFF -DENABLE_OV_PYTORCH_FRONTEND=ON -DENABLE_OV_JAX_FRONTEND=OFF -DENABLE_OV_IR_FRONTEND=ON \
+              -DENABLE_CLDNN=OFF -DENABLE_INTEL_GPU=OFF -DENABLE_INTEL_NPU=OFF -DENABLE_HETERO=OFF -DENABLE_AUTO=OFF -DENABLE_AUTO_BATCH=OFF -DENABLE_MULTI=OFF -DENABLE_PROXY=OFF \
+              -DENABLE_FASTER_BUILD=y \
+              -DCPACK_GENERATOR=NPM \
+              -DENABLE_PYTHON=OFF \
+              -DENABLE_WHEEL=OFF \
+              -DCPACK_PACKAGE_FILE_NAME=genai_nodejs_bindings \
+              -S ./openvino -B ./build
+          cmake --build ./build --target package -j
+
+      - name: Run javascript tests
+        working-directory: ${{ env.GENAI_REPO }}/src/js
+        run: |
+          mkdir bin
+          cp ${{ env.BUILD_DIR }}/genai_nodejs_bindings.tar.gz bin
+          cd bin && tar -xzf genai_nodejs_bindings.tar.gz && rm genai_nodejs_bindings.tar.gz && cd ..
+          npm install
+          npm test
+
+      #
+      # Upload build artifacts and logs
+      #
+
+      - name: Upload genai nodejs bindings archive
+        if: ${{ always() }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: genai_nodejs_bindings
+          path: ${{ env.BUILD_DIR }}/genai_nodejs_bindings.tar.gz
+          if-no-files-found: 'error'
+
+  genai_nodejs_samples_tests:
```

Comment:
can we perform samples tests within a single JS related job which builds GenAI JS and performs all the tests ?

Reply:
Done